### PR TITLE
Register as update driver for rpm-ostree

### PR DESCRIPTION
**Pull request by kelvinfan001**
Following https://github.com/coreos/rpm-ostree/pull/2459, rpm-ostree
now has a mechanism for third-party update drivers to register as
its update driver.

Register Zincati as the update driver in the "initialization" stage.

### CODE REVIEW

File: src/rpm_ostree/cli_deploy.rs
Comment (lucab) [inline src/rpm_ostree/cli_deploy.rs:54]: Hmm, it looks like this argument should be optional/conditional maybe?

File: src/rpm_ostree/cli_deploy.rs
Comment (lucab) [inline src/rpm_ostree/cli_deploy.rs:47]: It feels like this could be split into a `invoke_cli_deploy` plus a `invoke_cli_register`.

**Comment by lucab** [comment]
Thanks, looks sane overall. I guess we'd have to wait for an rpm-ostree release with the new API before landing this.

**Comment by lucab** [comment]
Additionally, can you check what's the resulting behavior when configuration contains https://coreos.github.io/zincati/usage/auto-updates/#disabling-auto-updates? Does `rpm-ostree status` output matches your expectation?

File: src/rpm_ostree/cli_deploy.rs
Comment (kelvinfan001) [reply src/rpm_ostree/cli_deploy.rs:47]: Sure! I was thinking that they share mostly the same code, but I guess forcing them together like this does feel a bit unnatural.

**Comment by kelvinfan001** [comment]
> can you check what's the resulting behavior when configuration contains https://coreos.github.io/zincati/usage/auto-updates/#disabling-auto-updates?

My expectation is that we still consider Zincati to be the update driver even when Zincati's auto-updates logic is disabled by configuration. After https://github.com/coreos/zincati/issues/94, I'm planning on getting rpm-ostree to check the systemd status of its update driver and also display that status in `rpm-ostree status`. So when Zincati's present on the system but its auto-updates logic is disabled, `rpm-ostree status` might look something like this:
```
AutomaticUpdates: driven by Zincati
UpdateDriverStatus: auto-updates disabled
```

File: src/rpm_ostree/cli_deploy.rs
Comment (lucab) [reply src/rpm_ostree/cli_deploy.rs:47]: Yes, I got the same feeling while reading through. The fake `Release` item in particular is a good indicator that this arrangement could be re-calibrated.

**Comment by lucab** [comment]
@kelvinfan001 that seems reasonable to me. When we finally get to glue all the pieces together, let's do a final manual test to make sure the UX matches the plan.

**Comment by cgwalters** [comment]
v2021.2 is out https://github.com/coreos/rpm-ostree/releases/tag/v2021.2 and I'll add the overrides to FCOS next-devel soon.

File: src/update_agent/actor.rs
Comment (kelvinfan001) [inline src/update_agent/actor.rs:198]: I'm not sure if this is the right thing to do. We don't actually need to wait for the results of `local_deployments()` before we `register_as_driver()`, so chaining like this seems unnecessary. But OTOH, I'm not sure how to combine `local_deployments()` and `register_as_driver()` into one elegantly. 
@lucab

Comment (kelvinfan001) [reply src/update_agent/actor.rs:198]: Hmm or does all of this not really matter since we're really just chaining all the actions within a tick into one big future..

**Comment by kelvinfan001** [comment]
Minor change: we no longer register as the updates driver if auto-updates logic is disabled. This is so that rpm-ostree's CI can freely use `rpm-ostree upgrade`, `rpm-ostree deploy`, etc. without having to unregister/disable Zincati first (since I believe on non-production streams/dev machines Zincati is disabled only through Zincati's configuration, i.e. Zincati's initialization still happens).
@lucab Let me know if this logic doesn't make sense, though. Thanks!

**Comment by lucab** [comment]
@kelvinfan001 that's an interesting scenario. Yes I think it is correct to not register as a driver in that case.

**Review by lucab — APPROVED** [review]

**Pull request merged**
